frontend/src/components/Pagination/PaginationDesktop.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Desktop variant for the Pagination component.
+ */
+
+import { useCallback } from 'react'
+import { BiChevronLeft, BiChevronRight } from 'react-icons/bi'
+import {
+  Box,
+  Button,
+  HStack,
+  IconButton,
+  Text,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+import { PAGINATION_THEME_KEY } from '~theme/components/Pagination'
+import { usePaginationRange } from '~hooks/usePaginationRange'
+
+import { PaginationProps } from './Pagination'
+
+// Separate constant to denote a separator in the pagination component.
+const SEPARATOR = '\u2026'
+
+interface DesktopPageButtonProps {
+  selectedPage: PaginationProps['currentPage']
+  page: number | typeof SEPARATOR
+  onClick: PaginationProps['onPageChange']
+  isDisabled: PaginationProps['isDisabled']
+}
+
+const DesktopPageButton = ({
+  selectedPage,
+  page,
+  onClick,
+  isDisabled,
+}: DesktopPageButtonProps) => {
+  const isSelected = page === selectedPage
+
+  const styles = useMultiStyleConfig(PAGINATION_THEME_KEY, { isSelected })
+
+  const handleClick = useCallback(() => {
+    if (page === SEPARATOR) return
+    onClick(page)
+  }, [onClick, page])
+
+  if (page === SEPARATOR) {
+    return <Text sx={styles.separator}>{page}</Text>
+  }
+
+  return (
+    <Button sx={styles.button} onClick={handleClick} isDisabled={isDisabled}>
+      {page}
+    </Button>
+  )
+}
+
+export const PaginationDesktop = ({
+  siblingCount = 1,
+  pageSize,
+  onPageChange,
+  totalCount,
+  currentPage,
+  isDisabled,
+}: PaginationProps): JSX.Element => {
+  const paginationRange = usePaginationRange<typeof SEPARATOR>({
+    totalCount,
+    pageSize,
+    currentPage,
+    siblingCount,
+    separator: SEPARATOR,
+  })
+
+  const styles = useMultiStyleConfig(PAGINATION_THEME_KEY, {})
+
+  const totalPageCount = Math.ceil(totalCount / pageSize)
+  const isDisableNextPage = isDisabled || currentPage >= totalPageCount
+  const isDisablePrevPage = isDisabled || currentPage <= 1
+
+  const handlePageBack = useCallback(() => {
+    if (isDisablePrevPage) return
+    onPageChange(currentPage - 1)
```

Comment:
the component seems to rely on the parent keeping the state of `currentPage` consistent, i.e. there might be a case where the currently focused page is different from the `currentPage` prop. am I understanding that correctly, and might this be a source of bugs in future? an alternative might be to rename `currentPage` to `initialPage`, and the component to keep track of `currentPage` in its internal state (perhaps exposed to the parent via a ref?)

Reply:
This components follows how controlled components are usually used though; only rendering what the parent passes in. If onPageChange does not change the currentPage the currentPage will remain as is -- the parent component controls that.

(Similar to how `value` props and `onChange` props must be passed in for the value to get updated)

I think I will stick with the current implementation if there are no big objections.